internal/bminventory/inventory.go

```diff
@@ -5227,67 +5226,8 @@ func (b *bareMetalInventory) validateDNSDomain(cluster common.Cluster, params in
 	return nil
 }
 
-func ipAsUint(ipStr string, log logrus.FieldLogger) uint64 {
-	parts := strings.Split(ipStr, ".")
-	if len(parts) != 4 {
-		log.Warnf("Invalid ip %s", ipStr)
-		return 0
-	}
-	var result uint64 = 0
-	for _, p := range parts {
-		result = result << 8
-		converted, err := strconv.ParseUint(p, 10, 64)
-		if err != nil {
-			log.WithError(err).Warnf("Conversion of %s to uint", p)
-			return 0
-		}
-		result += converted
-	}
-	return result
-}
-
-func applyLimit(ret models.FreeAddressesList, limitParam *int64) models.FreeAddressesList {
-	if limitParam != nil && *limitParam >= 0 && *limitParam < int64(len(ret)) {
-		return ret[:*limitParam]
-	}
-	return ret
-}
-
-func (b *bareMetalInventory) getFreeAddresses(_ context.Context, params installer.GetFreeAddressesParams, log logrus.FieldLogger) (models.FreeAddressesList, error) {
-	var hosts []*models.Host
-	err := b.db.Select("free_addresses").Find(&hosts, "cluster_id = ? and status in (?)", params.ClusterID.String(), []string{models.HostStatusInsufficient, models.HostStatusKnown}).Error
-	if err != nil {
-		return nil, common.NewApiError(http.StatusInternalServerError, errors.Wrapf(err, "Error retreiving hosts for cluster %s", params.ClusterID.String()))
-	}
-	if len(hosts) == 0 {
-		return nil, common.NewApiError(http.StatusNotFound, errors.Errorf("No hosts where found for cluster %s", params.ClusterID))
-	}
-	resultingSet := network.MakeFreeAddressesSet(hosts, params.Network, params.Prefix, log)
-
-	ret := models.FreeAddressesList{}
-	for a := range resultingSet {
-		ret = append(ret, a)
-	}
-
-	// Sort addresses
-	sort.Slice(ret, func(i, j int) bool {
-		return ipAsUint(ret[i].String(), log) < ipAsUint(ret[j].String(), log)
-	})
-
-	ret = applyLimit(ret, params.Limit)
-
-	return ret, nil
-}
-
 func (b *bareMetalInventory) GetFreeAddresses(ctx context.Context, params installer.GetFreeAddressesParams) middleware.Responder {
-	log := logutil.FromContext(ctx, b.log)
-
-	results, err := b.getFreeAddresses(ctx, params, log)
-	if err != nil {
-		log.WithError(err).Warn("GetFreeAddresses")
-		return common.GenerateErrorResponder(err)
-	}
-	return installer.NewGetFreeAddressesOK().WithPayload(results)
+	return common.NewApiError(http.StatusNotImplemented, errors.New(APINotImplemented))
```

Comment:
I feel like this should probably be a 404.
A 5xx error usually indicates a server problem that can't be resolved by the client, but this is a client problem.
Additionally what is happening in this case doesn't really match the description of 501 (ref: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/501)

cc @avishayt what do you think?

Reply:
We did the same with the wrapper that you implemented so it make sense. @nmagnezi sorry if i confused you with that.

Reply:
Ack. will change to 404.